nilearn/decomposition/_base.py

```diff
@@ -382,6 +383,12 @@ def __init__(
         self.n_jobs = n_jobs
         self.verbose = verbose
 
+        # Does not support surface-based images yet; See #4756 for updates.
+        if isinstance(self.mask, (SurfaceMasker, SurfaceImage)):
+            raise NotImplementedError(
+                "Surface-based images are not yet supported by this module."
+            )
+
```

Comment:
Raise an error if surface mask is used with Parcellations

Reply:
can you put this check in the fit method?
scikit learn convention says we should not have input validation in the constructor: https://scikit-learn.org/stable/developers/develop.html#instantiation

it is a rule we have not applied systematically but I intend to start fixing things progressively on this front, so let's at least try to stick to those conventions in the new code we add

Reply:
- see https://github.com/nilearn/nilearn/issues/4538

Reply:
Done!